go-controller/pkg/clustermanager/egressip_controller.go

```diff
@@ -1202,11 +1278,40 @@ func (eIPC *egressIPClusterController) validateEgressIPSpec(name string, egressI
 			eIPC.recorder.Eventf(&eIPRef, v1.EventTypeWarning, "InvalidEgressIP", "egress IP: %s for object EgressIP: %s is not a valid IP address", egressIP, name)
 			return nil, fmt.Errorf("unable to parse provided EgressIP: %s, invalid", egressIP)
 		}
-		validatedEgressIPs.Insert(ip.String())
+		validatedEgressIPs.Insert(egressIP)
 	}
 	return validatedEgressIPs, nil
 }
 
+func (eIPC *egressIPClusterController) isEgressIPAddrConflict(egressIP net.IP) (bool, error) {
+	nodes, err := eIPC.watchFactory.GetNodes()
+	if err != nil {
+		return false, fmt.Errorf("failed to get nodes: %v", err)
+	}
+	for _, node := range nodes {
+		nodeHostAddressesSet, err := util.ParseNodeHostAddresses(node)
+		if err != nil {
+			return false, fmt.Errorf("failed to parse node host addresses")
+		}
+		for _, nodeHostAddressT := range nodeHostAddressesSet.UnsortedList() {
+			nodeHostAddress, _ := any(nodeHostAddressT).(string)
```

Comment:
```
func (s Set[T]) UnsortedList() []T {
```
It would work without casting but my IDE complains so I cast it to silence the error it shows. I am fine to remove if you wish - just say.